openedx/frontend-app-course-authoring
# feat: [FC-0044] Course unit - Copy/paste functionality

**Settings**

```yaml
EDX_PLATFORM_REPOSITORY: https://github.com/raccoongang/edx-platform.git
EDX_PLATFORM_VERSION: ts-develop

TUTOR_GROVE_WAFFLE_FLAGS:
  - name: contentstore.new_studio_mfe.use_new_unit_page
    everyone: true

TUTOR_GROVE_MFE_LMS_COMMON_SETTINGS:
  MFE_CONFIG:
    ENABLE_UNIT_PAGE: true
```

### Description

In this PR, the opportunity for the user to copy and insert xblocks and units was added.

The primary features were implemented:
- A new Copy/paste functionality.

**Issue:** https://github.com/openedx/platform-roadmap/issues/321

### Developer notes
- The initial 1 commit in this PR is temporary. Once the https://github.com/openedx/frontend-app-course-authoring/pull/882 is merged, the third commit of this PR will become the main commit.
- More details: https://github.com/openedx/edx-platform/pull/32346

### Testing instructions
1. Run master devstack.
2. Start platform make dev.up.lms+cms+frontend-app-course-authoring and make checkout on this branch.
3. Enable the new Unit page by adding a waffle flag `contentstore.new_studio_mfe.use_new_unit_page` in the CMS admin panel.
4. Make sure that the MFE setting `ENABLE_UNIT_PAGE=true` is enabled.
5. Enable `contentstore.enable_copy_paste_units` a waffle flag in the CMS admin panel.
6. Go to the Course Unit page from the Course Outline page.

https://github.com/openedx/frontend-app-course-authoring/assets/93188219/7d1362dd-e43a-45c2-9532-c45d7e4f02d5
